### PR TITLE
dependency_support: bump bazel_rules_hdl

### PR DESCRIPTION
**Pull request by lpawelcz**
This PR includes fixes for https://github.com/google/xls/pull/1031:

* improve hermetic python handling
* remove cpython and embedded_python_interpreter

Additionally, the bumped version of `bazel_rules_hdl` include work related to improving bazel place_and_route flow (https://github.com/hdl/bazel_rules_hdl/issues/239) and benchmarking experience (https://github.com/google/xls/issues/1226)

CC @proppy

### CODE REVIEW

File: dependency_support/rules_hdl/workspace.bzl
Comment (ericastor) [inline dependency_support/rules_hdl/workspace.bzl:40]: Please update this date!

**Review by ericastor — CHANGES_REQUESTED** [review]

File: dependency_support/rules_hdl/workspace.bzl
Comment (lpawelcz) [reply dependency_support/rules_hdl/workspace.bzl]: Right, thanks for pointing that out.

**Comment by lpawelcz** [comment]
There are some pending PRs in `bazel_rules_hdl` which are related to `place_and_route` flow and design benchmarking that should be merged in the nearest future, e.g.:
* https://github.com/hdl/bazel_rules_hdl/pull/247
* changes from https://github.com/hdl/bazel_rules_hdl/pull/243 split into a batch of small PRs based on [upstreaming plan](https://github.com/hdl/bazel_rules_hdl/pull/243#pullrequestreview-1781000193)

We could wait for these to get merged and then bump this PR to include all changes or merge as it is now and open another PR with next update once `bazel_rules_hdl` PRs are merged.

**Comment by lpawelcz** [comment]
@proppy all recent PRs with improvements to place and route rules in `bazel_rules_hdl` are merged. I've bumped our `bazel_rules_hdl` dependency to include those.

**Review by proppy — APPROVED** [review]

**Pull request merged**
